ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -224,6 +225,19 @@ private KsqlEntity executeStatement(
     }
   }
 
+  /**
+   * Validate the statement by creating the execution plan for it.
+   * 
+   * @param statement
+   * @param statementText
+   * @param streamsProperties
+   * @throws Exception
+   */
+  private void validateStatement(Statement statement, String statementText,
+                                 Map<String, Object> streamsProperties) throws Exception {
+    getStatementExecutionPlan(statement, statementText, streamsProperties);
```

Comment:
I am still confused about this. The `getStatementExecutionPlan` seems to just generate the full plan and returns an `ExecutionPlan`. If it fails, it throws an exception. What we are doing here is just running the whole process and then saying it's valid if no exception is thrown. I presume that the execution plan would have to be eventually generated again. This seems wasteful. 

Why can't we retain the returned execution plan and use it directly?

Reply:
Yes, we will generate the execution plan again but not at the same place but in the engine in every instance of the server. This validation is just executed in the server that has received the query from user. Also this validation does not alter the metastore while the other ones that we perform in the engines does.